dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -809,6 +809,17 @@ private void checkCanShareWorkflow(User user, Workflow workflow) {
         return this.permissionsInterface.getPermissionsForWorkflow(user, workflow);
     }
 
+    @GET
+    @Timed
+    @UnitOfWork
+    @Path("/path/workflow/{repository}/actions")
+    @ApiOperation(value = "Gets all actions a user can perform on a workflow", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, notes = "", response = Role.Action.class, responseContainer = "List")
+    public List<Role.Action> getWorkflowActions(@ApiParam(hidden = true) @Auth User user, @ApiParam(value = "repository path", required = true) @PathParam("repository") String path) {
+        Workflow workflow = workflowDAO.findByPath(path, false);
+        checkEntry(workflow);
```

Comment:
Should use checkUser as well.

Reply:
If you are not an owner, this will return an empty list. I don't think we want to call checkUser(), because it verifies that you're an owner, and this endpoint is intended for non-owners to invoke as well.